FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_IP.c

```diff
@@ -895,47 +895,49 @@ NetworkBufferDescriptor_t * pxNewBuffer;
 }
 /*-----------------------------------------------------------*/
 
-#if( ipconfigZERO_COPY_TX_DRIVER != 0 ) || ( ipconfigZERO_COPY_RX_DRIVER != 0 )
+NetworkBufferDescriptor_t *pxPacketBuffer_to_NetworkBuffer( const void *pvBuffer )
+{
+uintptr_t uxBuffer;
+NetworkBufferDescriptor_t *pxResult;
 
```

Comment:
These functions:
~~~c
NetworkBufferDescriptor_t *pxPacketBuffer_to_NetworkBuffer( const void *pvBuffer );
NetworkBufferDescriptor_t *pxUDPPayloadBuffer_to_NetworkBuffer( const void * pvBuffer );
~~~
are very similar Each one subtracts a number of bytes from a pointer, and reads the address of a `NetworkBufferDescriptor_t`.
Because of this similarity, I propose to have the second call the first.

Another solution would be to add a common function:
~~~c
#if( ipconfigZERO_COPY_TX_DRIVER != 0 ) || ( ipconfigZERO_COPY_RX_DRIVER != 0 )
    NetworkBufferDescriptor_t *pxPacketBuffer_to_NetworkBuffer( const void *pvBuffer )
    {
        return prvBuffer_to_NetworkBuffer( pvBuffer, 0U );
    }
#endif /* ( ipconfigZERO_COPY_TX_DRIVER != 0 ) || ( ipconfigZERO_COPY_RX_DRIVER != 0 ) */

NetworkBufferDescriptor_t *pxUDPPayloadBuffer_to_NetworkBuffer( const void * pvBuffer )
{
    return prvBuffer_to_NetworkBuffer( pvBuffer, sizeof( UDPPacket_t ) );
}
~~~
Does that look better?

Reply:
Yes indeed. Both solutions would work.
I personally like the second soulution though.